Cabal/Distribution/Simple/Register.hs

```diff
@@ -411,9 +411,31 @@ generalInstalledPackageInfo adjustRelIncDirs pkg abi_hash lib lbi clbi installDi
     IPI.importDirs         = [ libdir installDirs | hasModules ],
     -- Note. the libsubdir and datasubdir templates have already been expanded
     -- into libdir and datadir.
+    --
+    -- TODO: Right now, only GHC > 8.0.1 supports a package database entry with
+    -- a field that indicates where the dynamic libraries are installed. When
+    -- more compilers start to support this, we should drop the
+    -- `supportsLibraryDynDir` check here, and just handle `IPI.libraryDynDirs`
+    -- in the actual package registration function belonging to those particular
+    -- compilers.
```

Comment:
Unfortunately we cannot do this yet given the structure of the code. We need to be able to generate the InstalledPackageInfo so we can write it to a file or script, without ending up calling the compiler specific `registerPackage`. What would be nice is if we had a compiler specific bit of generating the registration info, then we could do this compat bit there.

FYI, doing that might be an alternative to the thing I suggest below about compiler version tests.